src/basic_types.jl

```diff
@@ -76,20 +76,31 @@ struct BoundaryConditionDoNothing end
 # This version can be called by hyperbolic solvers on logically Cartesian meshes
 @inline function (::BoundaryConditionDoNothing)(
     u_inner, orientation_or_normal_direction, direction::Integer, x, t, surface_flux, equations)
+
   return flux(u_inner, orientation_or_normal_direction, equations)
 end
 
 # This version can be called by hyperbolic solvers on unstructured, curved meshes
-@inline function (::BoundaryConditionDoNothing)(
-    u_inner, outward_direction::AbstractVector, x, t, surface_flux, equations)
+@inline function (::BoundaryConditionDoNothing)(u_inner, outward_direction::AbstractVector,
+                                                x, t, surface_flux, equations)
+
   return flux(u_inner, outward_direction, equations)
 end
 
+# This version is called when passing in a non-conservative surface flux
+@inline function (::BoundaryConditionDoNothing)(u_inner, outward_direction::AbstractVector,
+                                                x, t, using_nonconservative_flux::True,
+                                                nonconservative_surface_flux, equations)
+
+  return nonconservative_surface_flux(u_inner, u_inner, outward_direction,
+                                      outward_direction, equations)
+end
+
```

Comment:
Is this still needed? It does not seem to be covered by our CI tests

Reply:
Deleted based on convo in https://github.com/trixi-framework/Trixi.jl/pull/1431/files#r1192543174